track.go

```diff
@@ -1,76 +1,173 @@
 package webrtc
 
 import (
-	"crypto/rand"
-	"encoding/binary"
+	"fmt"
+	"sync"
 
-	"github.com/pions/rtcp"
 	"github.com/pions/rtp"
 	"github.com/pions/webrtc/pkg/media"
-	"github.com/pkg/errors"
 )
 
-// Track represents a track that is communicated
+const rtpOutboundMTU = 1400
+
+// Track represents a single media track
 type Track struct {
```

Comment:
If all of the members are unexposed, would it make sense for this to be `track` and make `Track` an interface? I'm not sure why, but that *feels* like the right thing to do here...

Reply:
Yea I don't have an issue with it!

Is the goal to make code easier for the reader? I don't think we will have an internal behavior differences. 

In the future we could also add different Track behaviors (people could pass their own also) which might be interesting!

Reply:
Yeah, i think the smaller we keep the public interface, the easier it is to read. Also, the "different track behaviors" idea is interesting (but i'm not sure how it would work, or why)

Reply:
You could then maybe create a `Track` constructor that automatically pulled from a webcam. So if someone called `camera.NewTrack()` it would return something that users don't have to do the read/write on at all. But would implement our interface.

Just spitballing, no idea if that would all work out (or even make sense)

Reply:
I had a pretty hard time making this happen, there are some things I don't want to export that would need to be to make this happen `determinePayloadType` and setting the values of members? Mind if we table this for now.

If you feel strongly I can try some other stuff with this!